modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,136 @@
+
+Before you `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Ingress Controller.
+* You must expose a route for your service and specify the `--hostname` argument for your Ingress Controller domain.
+* must preserve the non-default port in the curl in order to access, for example: curl <route>:<port> // what does this mean?
+* Ensure that Ingress Controller routes can access your {product-title} cluster through the host IP address assigned to each node.
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the Ingress Controller
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: custom-ic3 <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: my-custom-ic.ipi-cluster.example.com <2>
+    nodePlacement:
+            nodeSelector:
+              matchLabels:
+                ingresscontroller: "custom-ic3" <3>
+    namespaceSelector:
+     matchLabels:
+       ic: custom-ic3 <4>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the `name` with a name for the `IngressController` object.
+<2> Specify the `domain` for the wilcard DNS record.
+<3> Specify the label for the `ingresscontroller` CR.
+<4> Matches routes from any namespaces with this label.
+
+. Add a label to a node by using the `oc label node` command:
++
+[source,terminal]
+----
+$ oc label node <node_name> ingresscontroller=<ingress_controller_custom_resource>
+----
+
+. Create the `Pod` object:
++
+[source,terminal]
+----
+$ oc create -f <pod_object_name>.yaml
+----
+
+. Locate the hostname by running the `oc get svc` command against the `openshift-ingress` namespace. The hostname of the load balancer is the `EXTERNAL-IP` associated with the `router-default` service in the `openshift-ingress` namespace.
+
+[source,bash]
+----
+$ oc get svc -n openshift-ingress
+NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                     AGE
+router-internal-custom-ic   ClusterIP   172.30.177.55    <none>        80/TCP,443/TCP,1936/TCP                     155m
```

Comment:
I would remove this line.  You can expect to see the ingresscontroller you just created and the default one but this one, from a different example in my doc, just confuses things.

Reply:
:heavy_check_mark: